src/service/viewer-impl.js

```diff
@@ -203,9 +211,8 @@ export class Viewer {
         this.overtakeHistory_;
     log.fine(TAG_, '- history:', this.overtakeHistory_);
 
-    this.visibilityState_ = this.params_['visibilityState'] ||
-        this.visibilityState_;
-    log.fine(TAG_, '- visibilityState:', this.visibilityState_);
+    this.setVisibilityState_(this.params_['visibilityState'],
+        /* opt_fix_unloaded */ true, /* opt_skip_fire */ true);
```

Comment:
Is fixedUnloaded really `true` here? Seems like the initial `hidden` would indicate instead the prerendering state, not unloaded.